Modules/_ctypes/stgdict.c

```diff
@@ -337,6 +337,26 @@ MakeAnonFields(PyObject *type)
     return 0;
 }
 
+/*
+  Append {padding}x to the PEP3118 format string.
+*/
+char *
+_ctypes_alloc_format_padding(const char *prefix, Py_ssize_t padding)
+{
+    /* int64 decimal characters + x + null */
+    char buf[19 + 1 + 1];
+
+    assert(padding > 0);
+
+    if (padding == 1) {
+        /* Use x instead of 1x, for brevity */
+        return _ctypes_alloc_format_string(prefix, "x");
+    }
+
+    sprintf(buf, "%zdx", padding);
```

Comment:
Worth using `snprintf` here (partly to pacify static checkers)?
```suggestion
    PyOS_snprintf(buf, sizeof(buf), "%zdx", padding);
```

(Aside: I'm not sure we actually _need_ `PyOS_snprintf` these days; I suspect that `snprintf` is good enough on  modern platforms.)

Reply:
I guess probably an assert to demonstrate the buffer is the right size is a good idea, if we're going to use the version that truncates?
```suggestion
    int ret = PyOS_snprintf(buf, sizeof(buf), "%zdx", padding);
    assert(0 <= ret && ret < sizeof(buf));
```

Otherwise you're trading off a buffer overflow for silent incorrect behavior, neither of which are great.

Reply:
Yes, that sounds good to me.